src/simulate.jl

```diff
@@ -247,11 +247,10 @@ function _simulate!(
 
     d = m.resp.d
 
-    if length(β) ≠ length(m.feterm.piv)
-        padding = length(model.feterm.piv) - m.feterm.rank
-        append!(β, fill(-0.0, padding))
+    if length(β) == length(m.feterm.piv)
+        # unlike LMM, GLMM stores the truncated, pivoted vector directly
```

Comment:
Tangential to this PR but it could be convenient to have some kind of `pivot` accessor to retrieve this array.